### ISSUE TITLE
Weird ResourceNotFoundException; Serious Error?

### ISSUE THREAD

**Issue by Voidi** [schnatterer/musicbrainzws2-java#4]
I'm getting a strange kind of behaviour:
With `new Recording().lookUp("15c99a4b-4a30-49be-a129-9c5b472240da")` literally the only line of code and printing the result to StandartOutput i get the following below. 
The returning recording object works fine so far (i only need recording title and associated artist), i don't know if this a serious error or something else.
My internet connection and the MB API (tested via Postman / API test program) itself are working as expected.

`Okt. 07, 2021 2:53:03 PM org.musicbrainz.wsxml.impl.JDOMParserWs2 createArtist
WARNUNG: Unrecognised Artist attribute: type-id
Okt. 07, 2021 2:53:03 PM org.musicbrainz.wsxml.impl.JDOMParserWs2 createRecording
WARNUNG: Unrecognised Recording element: first-release-date
Okt. 07, 2021 2:53:03 PM org.musicbrainz.wsxml.impl.JDOMParserWs2 createArtist
WARNUNG: Unrecognised Artist attribute: type-id
Okt. 07, 2021 2:53:03 PM org.musicbrainz.wsxml.impl.JDOMParserWs2 createArtist
WARNUNG: Unrecognised Artist attribute: type-id
Okt. 07, 2021 2:53:03 PM org.musicbrainz.wsxml.impl.JDOMParserWs2 createArtist
WARNUNG: Unrecognised Artist attribute: type-id
Okt. 07, 2021 2:53:03 PM org.musicbrainz.wsxml.impl.JDOMParserWs2 createArtist
WARNUNG: Unrecognised Artist attribute: type-id
Okt. 07, 2021 2:53:03 PM org.musicbrainz.wsxml.impl.JDOMParserWs2 createArtist
WARNUNG: Unrecognised Artist attribute: type-id
Okt. 07, 2021 2:53:03 PM org.musicbrainz.wsxml.impl.JDOMParserWs2 createArtist
WARNUNG: Unrecognised Artist attribute: type-id
Okt. 07, 2021 2:53:03 PM org.musicbrainz.wsxml.impl.JDOMParserWs2 createArtist
WARNUNG: Unrecognised Artist attribute: type-id
Okt. 07, 2021 2:53:03 PM org.musicbrainz.wsxml.impl.JDOMParserWs2 createArtist
WARNUNG: Unrecognised Artist attribute: type-id
Okt. 07, 2021 2:53:03 PM org.musicbrainz.wsxml.impl.JDOMParserWs2 createArtist
WARNUNG: Unrecognised Artist attribute: type-id
Okt. 07, 2021 2:53:03 PM org.musicbrainz.wsxml.impl.JDOMParserWs2 createArtist
WARNUNG: Unrecognised Artist attribute: type-id
Okt. 07, 2021 2:53:03 PM org.musicbrainz.wsxml.impl.JDOMParserWs2 createArtist
WARNUNG: Unrecognised Artist attribute: type-id
Okt. 07, 2021 2:53:03 PM org.musicbrainz.wsxml.impl.JDOMParserWs2 createArtist
WARNUNG: Unrecognised Artist attribute: type-id
Okt. 07, 2021 2:53:03 PM org.musicbrainz.wsxml.impl.JDOMParserWs2 createArtist
WARNUNG: Unrecognised Artist attribute: type-id
Okt. 07, 2021 2:53:03 PM org.musicbrainz.wsxml.impl.JDOMParserWs2 createArtist
WARNUNG: Unrecognised Artist attribute: type-id
Okt. 07, 2021 2:53:03 PM org.musicbrainz.wsxml.impl.JDOMParserWs2 createArtist
WARNUNG: Unrecognised Artist attribute: type-id
Okt. 07, 2021 2:53:03 PM org.musicbrainz.wsxml.impl.JDOMParserWs2 createArtist
WARNUNG: Unrecognised Artist attribute: type-id
Okt. 07, 2021 2:53:03 PM org.musicbrainz.wsxml.impl.JDOMParserWs2 createArtist
WARNUNG: Unrecognised Artist attribute: type-id
Okt. 07, 2021 2:53:03 PM org.musicbrainz.wsxml.impl.JDOMParserWs2 createArtist
WARNUNG: Unrecognised Artist attribute: type-id
Okt. 07, 2021 2:53:03 PM org.musicbrainz.wsxml.impl.JDOMParserWs2 createArtist
WARNUNG: Unrecognised Artist attribute: type-id
Okt. 07, 2021 2:53:03 PM org.musicbrainz.wsxml.impl.JDOMParserWs2 createArtist
WARNUNG: Unrecognised Artist attribute: type-id
Okt. 07, 2021 2:53:03 PM org.musicbrainz.wsxml.impl.JDOMParserWs2 createArtist
WARNUNG: Unrecognised Artist attribute: type-id
Okt. 07, 2021 2:53:03 PM org.musicbrainz.wsxml.impl.JDOMParserWs2 createArtist
WARNUNG: Unrecognised Artist attribute: type-id
Okt. 07, 2021 2:53:03 PM org.musicbrainz.wsxml.impl.JDOMParserWs2 createArtist
WARNUNG: Unrecognised Artist attribute: type-id
Okt. 07, 2021 2:53:03 PM org.musicbrainz.wsxml.impl.JDOMParserWs2 createArtist
WARNUNG: Unrecognised Artist attribute: type-id
Okt. 07, 2021 2:53:03 PM org.musicbrainz.wsxml.impl.JDOMParserWs2 createArtist
WARNUNG: Unrecognised Artist attribute: type-id
Okt. 07, 2021 2:53:03 PM org.musicbrainz.wsxml.impl.JDOMParserWs2 createWork
WARNUNG: Unrecognised Work attribute: type-id
Okt. 07, 2021 2:53:03 PM org.musicbrainz.wsxml.impl.JDOMParserWs2 createWork
WARNUNG: Unrecognised Work element: language-list
Okt. 07, 2021 2:53:03 PM org.musicbrainz.wsxml.impl.JDOMParserWs2 createArtist
WARNUNG: Unrecognised Artist attribute: type-id
Okt. 07, 2021 2:53:03 PM org.musicbrainz.wsxml.impl.JDOMParserWs2 createArtist
WARNUNG: Unrecognised Artist attribute: type-id
Okt. 07, 2021 2:53:03 PM org.musicbrainz.wsxml.impl.JDOMParserWs2 createArtist
WARNUNG: Unrecognised Artist attribute: type-id
Okt. 07, 2021 2:53:03 PM org.musicbrainz.wsxml.impl.JDOMParserWs2 createArtist
WARNUNG: Unrecognised Artist attribute: type-id
org.musicbrainz.webservice.ResourceNotFoundException: Server response was: 404 Not Found
        at org.musicbrainz.webservice.impl.HttpClientWebServiceWs2.executeMethod(HttpClientWebServiceWs2.java:263)
        at org.musicbrainz.webservice.impl.HttpClientWebServiceWs2.doGet(HttpClientWebServiceWs2.java:163)
        at org.musicbrainz.webservice.DefaultWebServiceWs2.get(DefaultWebServiceWs2.java:205)
        at org.musicbrainz.query.QueryWs2.getFromWebService(QueryWs2.java:74)
        at org.musicbrainz.query.search.SearchWs2.getMetadata(SearchWs2.java:70)
        at org.musicbrainz.query.search.AnnotationSearchWs2.execQuery(AnnotationSearchWs2.java:79)
        at org.musicbrainz.query.search.AnnotationSearchWs2.getOnePage(AnnotationSearchWs2.java:66)
        at org.musicbrainz.query.search.AnnotationSearchWs2.getNextPage(AnnotationSearchWs2.java:45)
        at org.musicbrainz.query.search.AnnotationSearchWs2.getFirstPage(AnnotationSearchWs2.java:38)
        at org.musicbrainz.query.search.AnnotationSearchWs2.getFullList(AnnotationSearchWs2.java:26)
        at org.musicbrainz.query.search.readysearches.AnnotationSearchbyEntityId.getFullList(AnnotationSearchbyEntityId.java:41)
        at org.musicbrainz.model.entity.EntityWs2.getAnnotation(EntityWs2.java:251)
        at org.musicbrainz.controller.Controller.loadAnnotation(Controller.java:351)
        at org.musicbrainz.controller.Recording.lookUp(Recording.java:218)
        at name.voidi.openlyricsapiserver.Main.main(Main.java:20)
Wings of Despair
`

**Comment by schnatterer** [comment]
I've also seen these kinds of warnings. I presume MB returns more properties than the lib implements.

I cant place the 404, though. There is an exception in the log, but you still get a working object?

**Comment by Voidi** [comment]
Sorry, i wrote this a bit confusing: Yes it seems that a Exception is printed to log, and the returining object seems fine

This sample:
```
public static void main(String[] args) {
	try {
		RecordingWs2 recording = new Recording().lookUp("15c99a4b-4a30-49be-a129-9c5b472240da");
		System.out.println(recording.getTitle());
		System.out.println(recording.getArtistCreditString());
	} catch (MBWS2Exception e) {
	}
}
```

results in:
```
Okt. 08, 2021 3:15:03 PM org.musicbrainz.wsxml.impl.JDOMParserWs2 createArtist
WARNUNG: Unrecognised Artist attribute: type-id
Okt. 08, 2021 3:15:03 PM org.musicbrainz.wsxml.impl.JDOMParserWs2 createRecording
WARNUNG: Unrecognised Recording element: first-release-date
Okt. 08, 2021 3:15:03 PM org.musicbrainz.wsxml.impl.JDOMParserWs2 createArtist
WARNUNG: Unrecognised Artist attribute: type-id
Okt. 08, 2021 3:15:03 PM org.musicbrainz.wsxml.impl.JDOMParserWs2 createArtist
WARNUNG: Unrecognised Artist attribute: type-id
Okt. 08, 2021 3:15:03 PM org.musicbrainz.wsxml.impl.JDOMParserWs2 createArtist
WARNUNG: Unrecognised Artist attribute: type-id
Okt. 08, 2021 3:15:03 PM org.musicbrainz.wsxml.impl.JDOMParserWs2 createArtist
WARNUNG: Unrecognised Artist attribute: type-id
Okt. 08, 2021 3:15:03 PM org.musicbrainz.wsxml.impl.JDOMParserWs2 createArtist
WARNUNG: Unrecognised Artist attribute: type-id
Okt. 08, 2021 3:15:03 PM org.musicbrainz.wsxml.impl.JDOMParserWs2 createArtist
WARNUNG: Unrecognised Artist attribute: type-id
Okt. 08, 2021 3:15:03 PM org.musicbrainz.wsxml.impl.JDOMParserWs2 createArtist
WARNUNG: Unrecognised Artist attribute: type-id
Okt. 08, 2021 3:15:03 PM org.musicbrainz.wsxml.impl.JDOMParserWs2 createArtist
WARNUNG: Unrecognised Artist attribute: type-id
Okt. 08, 2021 3:15:03 PM org.musicbrainz.wsxml.impl.JDOMParserWs2 createArtist
WARNUNG: Unrecognised Artist attribute: type-id
Okt. 08, 2021 3:15:03 PM org.musicbrainz.wsxml.impl.JDOMParserWs2 createArtist
WARNUNG: Unrecognised Artist attribute: type-id
Okt. 08, 2021 3:15:03 PM org.musicbrainz.wsxml.impl.JDOMParserWs2 createArtist
WARNUNG: Unrecognised Artist attribute: type-id
Okt. 08, 2021 3:15:03 PM org.musicbrainz.wsxml.impl.JDOMParserWs2 createArtist
WARNUNG: Unrecognised Artist attribute: type-id
Okt. 08, 2021 3:15:03 PM org.musicbrainz.wsxml.impl.JDOMParserWs2 createArtist
WARNUNG: Unrecognised Artist attribute: type-id
Okt. 08, 2021 3:15:03 PM org.musicbrainz.wsxml.impl.JDOMParserWs2 createArtist
WARNUNG: Unrecognised Artist attribute: type-id
Okt. 08, 2021 3:15:03 PM org.musicbrainz.wsxml.impl.JDOMParserWs2 createArtist
WARNUNG: Unrecognised Artist attribute: type-id
Okt. 08, 2021 3:15:03 PM org.musicbrainz.wsxml.impl.JDOMParserWs2 createArtist
WARNUNG: Unrecognised Artist attribute: type-id
Okt. 08, 2021 3:15:03 PM org.musicbrainz.wsxml.impl.JDOMParserWs2 createArtist
WARNUNG: Unrecognised Artist attribute: type-id
Okt. 08, 2021 3:15:03 PM org.musicbrainz.wsxml.impl.JDOMParserWs2 createArtist
WARNUNG: Unrecognised Artist attribute: type-id
Okt. 08, 2021 3:15:03 PM org.musicbrainz.wsxml.impl.JDOMParserWs2 createArtist
WARNUNG: Unrecognised Artist attribute: type-id
Okt. 08, 2021 3:15:03 PM org.musicbrainz.wsxml.impl.JDOMParserWs2 createArtist
WARNUNG: Unrecognised Artist attribute: type-id
Okt. 08, 2021 3:15:03 PM org.musicbrainz.wsxml.impl.JDOMParserWs2 createArtist
WARNUNG: Unrecognised Artist attribute: type-id
Okt. 08, 2021 3:15:03 PM org.musicbrainz.wsxml.impl.JDOMParserWs2 createArtist
WARNUNG: Unrecognised Artist attribute: type-id
Okt. 08, 2021 3:15:03 PM org.musicbrainz.wsxml.impl.JDOMParserWs2 createArtist
WARNUNG: Unrecognised Artist attribute: type-id
Okt. 08, 2021 3:15:03 PM org.musicbrainz.wsxml.impl.JDOMParserWs2 createArtist
WARNUNG: Unrecognised Artist attribute: type-id
Okt. 08, 2021 3:15:03 PM org.musicbrainz.wsxml.impl.JDOMParserWs2 createArtist
WARNUNG: Unrecognised Artist attribute: type-id
Okt. 08, 2021 3:15:03 PM org.musicbrainz.wsxml.impl.JDOMParserWs2 createWork
WARNUNG: Unrecognised Work attribute: type-id
Okt. 08, 2021 3:15:03 PM org.musicbrainz.wsxml.impl.JDOMParserWs2 createWork
WARNUNG: Unrecognised Work element: language-list
Okt. 08, 2021 3:15:03 PM org.musicbrainz.wsxml.impl.JDOMParserWs2 createArtist
WARNUNG: Unrecognised Artist attribute: type-id
Okt. 08, 2021 3:15:03 PM org.musicbrainz.wsxml.impl.JDOMParserWs2 createArtist
WARNUNG: Unrecognised Artist attribute: type-id
Okt. 08, 2021 3:15:03 PM org.musicbrainz.wsxml.impl.JDOMParserWs2 createArtist
WARNUNG: Unrecognised Artist attribute: type-id
Okt. 08, 2021 3:15:03 PM org.musicbrainz.wsxml.impl.JDOMParserWs2 createArtist
WARNUNG: Unrecognised Artist attribute: type-id
org.musicbrainz.webservice.ResourceNotFoundException: Server response was: 404 Not Found
        at org.musicbrainz.webservice.impl.HttpClientWebServiceWs2.executeMethod(HttpClientWebServiceWs2.java:263)
        at org.musicbrainz.webservice.impl.HttpClientWebServiceWs2.doGet(HttpClientWebServiceWs2.java:163)
        at org.musicbrainz.webservice.DefaultWebServiceWs2.get(DefaultWebServiceWs2.java:205)
        at org.musicbrainz.query.QueryWs2.getFromWebService(QueryWs2.java:74)
        at org.musicbrainz.query.search.SearchWs2.getMetadata(SearchWs2.java:70)
        at org.musicbrainz.query.search.AnnotationSearchWs2.execQuery(AnnotationSearchWs2.java:79)
        at org.musicbrainz.query.search.AnnotationSearchWs2.getOnePage(AnnotationSearchWs2.java:66)
        at org.musicbrainz.query.search.AnnotationSearchWs2.getNextPage(AnnotationSearchWs2.java:45)
        at org.musicbrainz.query.search.AnnotationSearchWs2.getFirstPage(AnnotationSearchWs2.java:38)
        at org.musicbrainz.query.search.AnnotationSearchWs2.getFullList(AnnotationSearchWs2.java:26)
        at org.musicbrainz.query.search.readysearches.AnnotationSearchbyEntityId.getFullList(AnnotationSearchbyEntityId.java:41)
        at org.musicbrainz.model.entity.EntityWs2.getAnnotation(EntityWs2.java:251)
        at org.musicbrainz.controller.Controller.loadAnnotation(Controller.java:351)
        at org.musicbrainz.controller.Recording.lookUp(Recording.java:218)
        at name.voidi.openlyricsapiserver.Main.main(Main.java:19)
Wings of Despair
Kamelot

````

**Comment by schnatterer** [comment]
I had a look at whats happening. For some reason the lib tries to find annotations for the recordings via this URL: https://search.musicbrainz.org/ws/2/annotation/?limit=100&offset=0&query=entity%3A15c99a4b-4a30-49be-a129-9c5b472240da
That's 404, hence the exception.
I never worked with recordings, though. The exceptions seems not to occur with releases.

I cant put any more efforts into this, but if you want to fix this error, I can offer review and releasing.

**Comment by Voidi** [comment]
With the help of the Musicbrainz Community a part seems solved [Musicbrainz Forum](https://community.metabrainz.org/t/get-annotations-from-the-api/554738)

The Request to `search.musicbrainz.org` seems remaining parts to old API function, and can safe changed to the main domain.
I tested this locally with code and no exception is thrown. Some warnings a printed to the log, but this will require further investigation.

**Comment by schnatterer** [comment]
Closed by #5
